bql/execution/evaluators.go

```diff
@@ -914,8 +923,11 @@ type funcApp struct {
 	fVal        reflect.Value
 	params      []Evaluator
 	paramValues []reflect.Value
+	selector    data.Path
 }
 
+const tmpKeyName = "tmpkey"
```

Comment:
Rename this to `tmpFuncAppSelectorKeyName` and add comment:

```
// tmpFuncAppSelectorKeyName is used to apply a selector to the result of a function.
// This has to be global because a selector is compiled in FuncAppSelector and a
// temporary map to perform data.Map.Get in funcApp.Eval has to have the same key.
// This temporary variable can be removed once data.Array supports Get(Path).
```

Reply:
You can merge #158 first and then use `Array.Get` in this PR to get rid of `tmpKeyName`.

Reply:
applied Array.Get to get selected values, thx!